design/ceph/object/store.md

```diff
@@ -351,3 +351,285 @@ spec:
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+If needed, create additional pool to use it in custom StorageClass, for example `GLACIER` and `REDUCED_REDUNDANCY`:
+> [!NOTE]  
+> Ceph allows arbitrary name for StorageClasses, however some clients/libs insist on AWS names so it is recommended to use one of the [valid x-amz-storage-class values](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax) for better compatibility:
+>
+> `STANDARD | REDUCED_REDUNDANCY | STANDARD_IA | ONEZONE_IA | INTELLIGENT_TIERING | GLACIER | DEEP_ARCHIVE | OUTPOSTS | GLACIER_IR | SNOW | EXPRESS_ONEZONE`
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: reduced.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+```
+
+Then update `CephObjectStore` CRD  `sharedPools` section with target placements and StorageClasses definitions:
+
+```yaml
+spec:
+  # [OPTIONAL] - sharedPools is mutually exclusive with spec.metadataPool and spec.dataPool
+  # It is the place to tell RGW which pools can be used to store buckets and objects data and metadata.
+  # All pools under this section should be created beforehand.
+  # Rook will append rados namespace to each of the pools, so it is safe to use the same pool across multiple RGW instances.
+  # It is also safe to use the same pool in different target placements.
+  sharedPools: 
+    #  [REQUIRED] - metadata pool. Will be used by default to store bucket index
+    metadataPoolName: default.rgw.buckets.index # results to default.rgw.buckets.index:<rgw_name>.index
+
+    #  [REQUIRED] - data pool. Will be used by default to store bucket objects
+    dataPoolName: default.rgw.buckets.data # results to default.rgw.buckets.data:<rgw_name>.data
+
+    #  [OPTIONAL] - non erasure coded data pool. Will be used by default to store bucket multipart objects
+    # If not set, dataPoolName will be used
+    dataNonECPoolName: "" # results to default.rgw.buckets.data:<rgw_name>.data-nonec
+
+    # [OPTIONAL] - Storage classes for default placement.
+    storageClasses: 
+      - name: GLACIER # [REQUIRED] - can be arbitrary but it is better to use one of AWS x-amz-storage-class names.
+        # [REQUIRED] overrides default dataPoolName for object with given StorageClass.
+        dataPoolName: glacier.rgw.buckets.data # results to glacier.rgw.buckets.data:<rgw_name>.glacier
+
+    # [OPTIONAL] - additional pool placement targets.
+    additional: 
+      # placement target named "cold"
+      cold:                                         # key will be used for placement target name. Name can be arbitrary.
+        metadataPoolName: cold.rgw.buckets.index    # [REQUIRED] results to cold.rgw.buckets.index:<rgw_name>.cold.index
```

Comment:
The RADOS namespace would just be the `<rgw_name>`, right? It's already in the specific pool, so I don't think we need the `cold.index` name on the rados namespace.
```suggestion
        metadataPoolName: cold.rgw.buckets.index    # [REQUIRED] results to cold.rgw.buckets.index:<rgw_name>
```

Reply:
yes, makes sense. i think that we don't need to append target name here even if the same pool will be used for multiple targets. 

It is probably not the best idea to use the same pool for index and data. But probably we can still add pool type to namespace just in case if someone will do so: `<pool_name>:<rgw_name>.index` and `<pool_name>:<rgw_name>.data`. What do you think?

Reply:
I agree it makes sense to do our best to ensure users can't configure their input in any way that causes aliased outputs, even if the input combos don't make sense. I'm often surprised by what users will try to do, with good reasoning that I couldn't imagine when the feature was created.

While in the context written where the pool name is `cold.rgw.buckets.index`, it seems silly. But I can come up with a case where it might matter. 

E.g., I might make a more generic pool name like `replica-2-fast` that I can use however I need to, without specific intent. In that context, it seems reasonable to me that GLACIER placement might use `replica-3-fast` for indexing at the same time STANDARD placement might use `replica-3-fast` for data. STANDARD might also use `replica-3-fast` for data as well. In this case, Rook would do best to ensure none of the chosen output pools/namespaces can conflict.